ribbon/src/test/java/feign/ribbon/LBClientTest.java

```diff
@@ -0,0 +1,56 @@
+package feign.ribbon;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
+import feign.Request;
+import feign.ribbon.LBClient.RibbonRequest;
+
+public class LBClientTest {
+
+  @Test
+  public void testRibbonRequest() throws URISyntaxException {
+    // test for RibbonRequest.toRequest()
+    // the url has a query whose value is an encoded json string
+    String urlWithEncodedJson = "http://test.feign.com/p?q=%7b%22a%22%3a1%7d";
+    URI uri = new URI(urlWithEncodedJson);
+    Map<String, Collection<String>> headers = new HashMap<String, Collection<String>>();
+    // create a Request as baseline for recreate and assert
+    Request requestOrign = Request.create("GET", uri.toASCIIString(), headers, null, Charset.forName("utf-8")); 
+    RibbonRequest ribbonRequest = new RibbonRequest(null, requestOrign, uri);
+
+    // use toRequest() recreate a Request
+    Request requestGetFormToRequest = ribbonRequest.toRequest();
+
+    // test the same attributes
+    assertEquals(requestOrign.method(), requestGetFormToRequest.method());
```

Comment:
this test is kindof hard to read. Can you look at other unit tests? you'll notice we use assertj which is helpful